skimage/_shared/testing.py

```diff
@@ -220,27 +221,39 @@ def test_parallel(num_threads=2):
     num_threads : int, optional
         The number of times the function is run in parallel.
 
+    warnings_matching: list or None
+        This parameter is passed on to `expected_warnings` so as not to have
+        race conditions with the warnings filters. A single
+        `expected_warnings` context manager is used for all threads.
+        If None, then no warnings are checked.
+
     """
 
     assert num_threads > 0
 
     def wrapper(func):
         @functools.wraps(func)
         def inner(*args, **kwargs):
-            threads = []
-            for i in range(num_threads - 1):
-                thread = threading.Thread(target=func, args=args,
-                                          kwargs=kwargs)
-                threads.append(thread)
-            for thread in threads:
-                thread.start()
-
-            result = func(*args, **kwargs)
-
-            for thread in threads:
-                thread.join()
-
-            return result
+            if warnings_matching is None:
+                # dummy context manager
+                ctx = memoryview(b'')
+            else:
+                ctx = expected_warnings(warnings_matching)
+            with ctx:
```

Comment:
This works??? Can you add links to documentation about this (the dummy context) in a comment? Because my mind is being blown.

Additionally, perhaps we can make expected warnings behave correctly with None input, which would simplify this code dramatically?

Reply:
https://stackoverflow.com/a/54786924

I swear I benchmarked this and memory view as the fastest nullcontext manager. I don't understand why python 3.7 chose anything else. Maybe they offer more debugging messages??

Reply:
What behaviour do you want? the no-op if None is passed?

Reply:
Yes: basically make `expected warnings` return that null memoryview when None is the input.

Reply:
This is in contrast to the behaviour that is done now with `[None]`, i forget what that does.

Reply:
i think it jsut skips over them.